### PR TITLE
feat(plasm-*): Bump typescript version to 4.0.8

### PR DESCRIPTION
**Pull request by neretin-trike**
Поднял typescript до мажорной 4 версии. Также обновил react-script до 4 версии, и переименовал некоторые правила в соответствии с документацией typescript-eslint - https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0. 

Временно отключил некоторые eslint/typescript правила

:baby_chick: Download canary assets:

[colors--canary.408.4414778437.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.408.4414778437.xml)  
[color_metro_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.408.4414778437.swift)  
[color_metro_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.408.4414778437.kt)  
[color_metro_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.408.4414778437.ts)  
[color_sberHealth_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.408.4414778437.swift)  
[color_sberHealth_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.408.4414778437.kt)  
[color_sberHealth_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.408.4414778437.ts)  
[color_sbermarket_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.408.4414778437.swift)  
[color_sbermarket_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.408.4414778437.kt)  
[color_sbermarket_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.408.4414778437.ts)  
[color_sberprime_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.408.4414778437.swift)  
[color_sberprime_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.408.4414778437.kt)  
[color_sberprime_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.408.4414778437.ts)  
[color_selgros_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.408.4414778437.swift)  
[color_selgros_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.408.4414778437.kt)  
[color_selgros_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.408.4414778437.ts)  
[color_smbusiness_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.408.4414778437.swift)  
[color_smbusiness_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.408.4414778437.kt)  
[color_smbusiness_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.408.4414778437.ts)  
[PlasmaTokensColor--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.408.4414778437.swift)  
[shadow_sbermarket_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.408.4414778437.ts)  
[typo_mage_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.408.4414778437.swift)  
[typo_mage_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.408.4414778437.kt)  
[typo_mage_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.408.4414778437.ts)  
[typo_plasma_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.408.4414778437.swift)  
[typo_plasma_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.408.4414778437.kt)  
[typo_plasma_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.408.4414778437.ts)  
[typo_ruler_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.408.4414778437.swift)  
[typo_ruler_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.408.4414778437.kt)  
[typo_ruler_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.408.4414778437.ts)  
[typo_sage_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.408.4414778437.swift)  
[typo_sage_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.408.4414778437.kt)  
[typo_sage_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.408.4414778437.ts)  
[typo_sbermarket_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.408.4414778437.swift)  
[typo_sbermarket_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.408.4414778437.kt)  
[typo_sbermarket_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.408.4414778437.ts)  
[typo_soulmate_ios-swift--canary.408.4414778437.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.408.4414778437.swift)  
[typo_soulmate_kotlin--canary.408.4414778437.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.408.4414778437.kt)  
[typo_soulmate_react-native--canary.408.4414778437.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.408.4414778437.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.157.0-canary.408.4414778437.0
  npm install @salutejs/plasma-colors@0.4.0-canary.408.4414778437.0
  npm install @salutejs/plasma-core@1.102.0-canary.408.4414778437.0
  npm install @salutejs/plasma-hope@0.15.0-canary.408.4414778437.0
  npm install @salutejs/plasma-icons@1.130.0-canary.408.4414778437.0
  npm install @salutejs/plasma-temple@1.141.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-b2b@1.11.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-b2c@0.20.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-core@0.2.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-web@1.26.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens@1.40.0-canary.408.4414778437.0
  npm install @salutejs/plasma-typo@0.33.0-canary.408.4414778437.0
  npm install @salutejs/plasma-ui@1.174.0-canary.408.4414778437.0
  npm install @salutejs/plasma-web@1.181.0-canary.408.4414778437.0
  npm install @salutejs/plasma-cy-utils@0.47.0-canary.408.4414778437.0
  npm install @salutejs/plasma-sb-utils@0.100.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-android@2.47.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-ios-swift@2.47.0-canary.408.4414778437.0
  npm install @salutejs/plasma-tokens-utils@0.19.0-canary.408.4414778437.0
  # or 
  yarn add @salutejs/plasma-b2c@1.157.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-colors@0.4.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-core@1.102.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-hope@0.15.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-icons@1.130.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-temple@1.141.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-b2b@1.11.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-b2c@0.20.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-core@0.2.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-web@1.26.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens@1.40.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-typo@0.33.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-ui@1.174.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-web@1.181.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-cy-utils@0.47.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-sb-utils@0.100.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-android@2.47.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.47.0-canary.408.4414778437.0
  yarn add @salutejs/plasma-tokens-utils@0.19.0-canary.408.4414778437.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-408/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-408/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-408/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-408/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-408/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-408/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-408/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-408/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-408/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-408/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-408/

File: packages/plasma-temple/package.json [hunk 1]
```diff
@@ -51,12 +51,12 @@
         "jest-config": "26.6.3",
         "react": "18.2.0",
         "react-dom": "18.2.0",
-        "react-scripts": "^3.4.4",
+        "react-scripts": "4.0.3",
```
Comment (Yeti-or) [inline packages/plasma-temple/package.json:54]: а почему?
в целом мы это же выпиливаем с вите да?

File: .eslintrc.js [hunk 2]
```diff
@@ -58,6 +58,19 @@ module.exports = {
 
         '@typescript-eslint/explicit-function-return-type': 'off',
 
+        // TODO: Удалить правила с 62 по 72, после влития https://github.com/salute-developers/plasma/pull/383
```
Comment (Yeti-or) [inline .eslintrc.js:61]: а почему сборщик влияет на правила eslint ?

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-temple/package.json (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-temple/package.json:54]: да, иначе сторибук не собирается на вебпаке с этой версией тайпскрипта

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-408/

File: .eslintrc.js (hunk 2 above)
Comment (neretin-trike) [reply .eslintrc.js:61]: хороший вопрос, но при сборке в vite без react-script и typescript~4.0.0 у меня такой проблемы не было. Всё собиралось для plasma-web и plasma-b2c нормально. Хотя я не проверял для plasma-temple и plasma-ui, мб там такие же проблемы возникли бы. Это уже можно будет посмотреть после влития этой прки

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-408/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-408/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-408/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-408/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-408/temple-storybook/

File: .eslintrc.js (hunk 2 above)
Comment (Yeti-or) [reply .eslintrc.js:61]: okay надо будет задачку завести на причесывание линтеров

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**